lib/src/codec/decoders/coap_message_decoder.dart

```diff
@@ -63,10 +77,10 @@ abstract class CoapMessageDecoder implements CoapIMessageDecoder {
 
   @override
   CoapResponse? decodeResponse() {
-    if (isResponse) {
-      final response = CoapResponse(code)
-        ..type = type
-        ..id = id;
+    final decodedCode = _decodedCode;
+    final coapMessageType = _decodedType;
+    if (decodedCode != null && coapMessageType != null && isResponse) {
```

Comment:
Tiny nitpick, I guess we could exit early here:
```dart
if (!isResponse || decodedCode == null || coapMessageType == null) {
  return null;
}
// ...
```

Won't make a big difference here, but defensive programming and early returns is easier to read, so less error prone I think. One discussion can be found here: https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement

Reply:
Oh, yeah, that is much better, thanks, I updated the code accordingly :)